packages/fcl-core/src/interaction-template-utils/derive-cadence-by-network/derive-cadence-by-network.test.js

```diff
@@ -0,0 +1,224 @@
+          FungibleToken: {
+            mainnet: {
+              address: "0xf233dcee88fe0abe",
+              fq_address: "A.0xf233dcee88fe0abe.FungibleToken",
+              contract: "FungibleToken",
+              pin: "83c9e3d61d3b5ebf24356a9f17b5b57b12d6d56547abc73e05f820a0ae7d9cf5",
+              pin_block_height: 34166296,
+            },
+            testnet: {
+              address: "0x9a0766d93b6608b7",
+              fq_address: "A.0x9a0766d93b6608b7.FungibleToken",
+              contract: "FungibleToken",
+              pin: "83c9e3d61d3b5ebf24356a9f17b5b57b12d6d56547abc73e05f820a0ae7d9cf5",
+              pin_block_height: 74776482,
+            },
+          },
+        },
+      },
+      arguments: {},
+    },
+  }
+
+  test("It derives cadence correctly for a given network", async () => {
+    let cadence = await deriveCadenceByNetwork({
+      network: "mainnet",
+      template,
+    })
+
+    expect(cadence).toEqual("import FungibleToken from 0xf233dcee88fe0abe\n")
+  })
+
+  test("It fails to derive cadence for unknown network", async () => {
+    await expect(() =>
+      deriveCadenceByNetwork({
+        network: "randomnet",
+        template,
+      })
+    ).rejects.toThrow(Error)
+  })
+})
+
+describe("Derive cadence by network 1.1.0", () => {
+  const template11 = {
+    f_type: "InteractionTemplate",
+    f_version: "1.1.0",
+    id: "3a99af243b85f3f6af28304af2ed53a37fb913782b3efc483e6f0162a47720a0",
+    data: {
+      type: "transaction",
+      interface: "",
+      messages: [
+        {
+          key: "title",
+          i18n: [
+            {
+              tag: "en-US",
+              translation: "Transfer Tokens",
+            },
+          ],
+        },
+        {
+          key: "description",
+          i18n: [
+            {
+              tag: "en-US",
+              translation: "Transfer Flow to account",
+            },
+          ],
+        },
+      ],
+      cadence: {
+        body: 'import "FungibleToken"\n\n#interaction(\n    version: "1.1.0",\n    title: "Transfer Flow",\n    description: "Transfer Flow to account",\n    language: "en-US",\n    parameters: [\n        Parameter(\n            name: "amount", \n            title: "Amount", \n            description: "The amount of FLOW tokens to send"\n        ),\n        Parameter(\n            name: "to", \n            title: "To",\n            description: "The Flow account the tokens will go to"\n        )\n    ],\n)\n\ntransaction(amount: UFix64, to: Address) {\n    let vault: @FungibleToken.Vault\n    \n    prepare(signer: AuthAccount) {\n        self.vault \u003c- signer\n            .borrow\u003c\u0026{FungibleToken.Provider}\u003e(from: /storage/flowTokenVault)!\n            .withdraw(amount: amount)\n    }\n\n    execute {\n        getAccount(to)\n            .getCapability(/public/flowTokenReceiver)!\n            .borrow\u003c\u0026{FungibleToken.Receiver}\u003e()!\n            .deposit(from: \u003c-self.vault)\n    }\n}',
```

Comment:
is there any way to format it in a more readable way? probably even use `.replace` for `\n`?

Reply:
This is how the template is generated. Not really meant to be readable.